tortoise/verifying_tortoise_test.go

```diff
@@ -3107,3 +3056,100 @@ func TestVoteBlockFilterForHealing(t *testing.T) {
 	assert.True(t, filter(goodBlock.ID()))
 	assert.True(t, filter(badBlock.ID()))
 }
+
+// gapVote will skip one layer in voting.
+func gapVote(rng *mrand.Rand, layers []*types.Layer) sim.Voting {
+	if len(layers) == 1 {
+		panic("need atleast 2 layers")
+	}
+	baseLayer := layers[len(layers)-2]
+	support := layers[len(layers)-2].BlocksIDs()
+	base := baseLayer.Blocks()[rng.Intn(len(baseLayer.Blocks()))]
+	return sim.Voting{Base: base.ID(), Support: support}
+}
+
+// olderExceptions will vote for block older then base block.
+func olderExceptions(rng *mrand.Rand, layers []*types.Layer) sim.Voting {
+	if len(layers) == 1 {
+		panic("need atleast 2 layers")
+	}
+	baseLayer := layers[len(layers)-1]
+	base := baseLayer.Blocks()[rng.Intn(len(baseLayer.Blocks()))]
+	voting := sim.Voting{Base: base.ID()}
+	for _, layer := range layers[len(layers)-2:] {
+		for _, bid := range layer.BlocksIDs() {
+			voting.Support = append(voting.Support, bid)
```

Comment:
this is not how it works, imagine that you asked tortoise to produce base blocks at layer 12 and it only has blocks from 10, how it can add votes for what it doesn't have?

Reply:
sorry, i somehow misplaced my comments.

in this case it doesn't matter if the exception is artificial or not, it is just to mark this block not good

Reply:
i think it matters for the reader. and it takes so little effort to make it correct, changing it from support to against, why not?

Reply:
the thing is that this is exactly how votes looked when we selected a base block that had a "gap" votes. 

example, 
block A from layer 12 uses base block B from layer 10, and no exceptions for layer 11 - this is block with gap vote, it is still good.
block C from layer 13 uses block A as base block, but because block A had to exceptions for layer 10 we have to add them ourselves, so block C will have exceptions for 12 and 11.

Reply:
i totally misunderstood what gap vote is supposed to be. i thought its when hare hasn't terminate for layer 11 when generating blocks for layer 12.